eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/FastList.java

```diff
@@ -262,18 +263,33 @@ public int getBatchCount(int batchSize)
         return this.toArray((E[]) new Object[sourceToIndex - sourceFromIndex + 1], sourceFromIndex, sourceToIndex, 0);
     }
 
+    /**
+     * Overrides default method from List.
+     *
+     * @since 10.0
+     */
+    @Override
+    public void sort(Comparator<? super T> comparator)
+    {
+        if (this.size > 1)
+        {
+            Arrays.sort(this.items, 0, this.size, comparator);
+        }
+    }
+
+    /**
+     * @since 10.0
+     */
     @Override
     public FastList<T> sortThis(Comparator<? super T> comparator)
     {
-        Arrays.sort(this.items, 0, this.size, comparator);
-        return this;
+        return (FastList<T>) super.sortThis(comparator);
```

Comment:
these overrides are because we return FastList. Should we break compatibility and replace it with a MutableList? It will be safe at runtime if users cast. We can then replace it by the default method.

Reply:
I think it's important not to break backwards compatibility here.  I was torn on whether to use the cast or reimplement the method as it is simple enough to do, but I didn't want to duplicate code, and the cast is safe here.

Reply:
👍